src/overlays/actors/ovl_Bg_Dy_Yoseizo/z_bg_dy_yoseizo.c

```diff
@@ -71,55 +64,979 @@ const ActorInit Bg_Dy_Yoseizo_InitVars = {
+            particleEnvColor.b = sParticleEnvColors[particleType].b;
+            BgDyYoseizo_ParticleInit(this, &particleInitPos, &particleInitVelocity, &particleAccel, &particlePrimColor,
+                                     &particleEnvColor, particleScale, particleLife, particleType);
+        }
+    }
+}
+
+void BgDyYoseizo_Bob(BgDyYoseizo* this, GlobalContext* globalCtx) {
+    this->targetHeight = this->grownHeight + this->bobOffset;
+    Math_ApproachF(&this->actor.world.pos.y, this->targetHeight, 0.1f, 10.0f);
+    Math_ApproachF(&this->bobOffset, 10.0f, 0.1f, 0.5f);
+
+    if (globalCtx->csCtx.state == 0) {
+        this->actor.velocity.y = Math_SinS(this->bobTimer);
+    } else {
+        this->actor.velocity.y = Math_SinS(this->bobTimer) * 0.4f;
+    }
+}
+
+void BgDyYoseizo_CheckMagicAcquired(BgDyYoseizo* this, GlobalContext* globalCtx) {
+    if (Flags_GetSwitch(globalCtx, 0x38)) {
+        globalCtx->msgCtx.unk_E3EE = 4;
+        if (globalCtx->sceneNum == SCENE_DAIYOUSEI_IZUMI) {
+            if (!(gSaveContext.magicAcquired) && (this->fountainType != FAIRY_UPGRADE_SPIN_ATTACK)) {
+                Actor_Kill(&this->actor);
+                return;
+            }
+        } else {
+            if (!(gSaveContext.magicAcquired)) {
+                Actor_Kill(&this->actor);
+                return;
+            }
+        }
+        func_8002DF54(globalCtx, &this->actor, 1);
+        this->actionFunc = BgDyYoseizo_ChooseType;
+    }
+}
+
+void BgDyYoseizo_ChooseType(BgDyYoseizo* this, GlobalContext* globalCtx) {
+    s32 givingReward;
+
+    func_8002DF54(globalCtx, &this->actor, 1);
+    // Mode
+    osSyncPrintf(VT_FGCOL(YELLOW) "☆☆☆☆☆ もうど ☆☆☆☆☆ %d\n" VT_RST, globalCtx->msgCtx.unk_E3EE);
+    givingReward = false;
+
+    if (globalCtx->sceneNum != SCENE_DAIYOUSEI_IZUMI) {
+        switch (this->fountainType) {
+            case FAIRY_SPELL_FARORES_WIND:
+                if (!(gSaveContext.itemGetInf[1] & 0x100)) {
+                    givingReward = true;
+                }
+                break;
+            case FAIRY_SPELL_DINS_FIRE:
+                if (!(gSaveContext.itemGetInf[1] & 0x200)) {
+                    givingReward = true;
+                }
+                break;
+            case FAIRY_SPELL_NAYRUS_LOVE:
+                if (!(gSaveContext.itemGetInf[1] & 0x400)) {
+                    givingReward = true;
+                }
+                break;
+        }
+    } else {
+        switch (this->fountainType) {
+            case FAIRY_UPGRADE_SPIN_ATTACK:
+                if (!(gSaveContext.magicAcquired) || BREG(2)) {
+                    // Spin Attack speed UP
+                    osSyncPrintf(VT_FGCOL(GREEN) " ☆☆☆☆☆ 回転切り速度ＵＰ ☆☆☆☆☆ \n" VT_RST, &gSaveContext);
+                    this->givingSpell = true;
+                    givingReward = true;
+                }
+                break;
+            case FAIRY_UPGRADE_DOUBLE_MAGIC:
+                if (!(gSaveContext.doubleMagic)) {
```

Comment:
same here with parens
you dont need to wrap them when it has multiple parts, like an & or something